### PR TITLE
Feat(eos_cli_config_gen): Support for configuring logging level facilities

### LINKED ISSUE #3740: Support for configuring logging facilities
### Enhancement summary

Possibility to configure different logging facilities, each of them with their corresponding severity level
Currently logging facilities (AAA, ACL, BFD, BGP, etc) cannot be specified/configured.

### Which component of AVD is impacted

eos_cli_config_gen

### Use case example

Configure logging for different facilities, the same way as it can be done with EOS;
logging level ARP informational
logging level BFD debug
logging level BGP notification

### Describe the solution you would like

eos_cli__config_gen should have a new string parameter under 'logging' variable, such as 'facilities', which could be configured with a specific severity level.
logging:
   facilities:
      - ARP: informational
      - BFD: debug
      - BGP: notification

### Describe alternatives you have considered

As an alternative, the following work-around can be used as of now:

custom_structured_configuration_logging:
eos_cli: |
   logging level ARP informational
   logging level BFD debug
   logging level BGP notifications

### Additional context

_No response_

### Contributing Guide

- [X] I agree to follow this project's Code of Conduct

### PR DESCRIPTION
**Pull request by MaheshGSLAB**
## Change Summary

Support for configuring logging level facilities. Added the function for logging level, facility and severity. 

## Related Issue(s)

Fixes #3740 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes
Added the support for logging level and severity on facility. 

## How to test
Run molecule

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2 [hunk 1]
```diff
@@ -93,4 +93,11 @@ logging source-interface {{ logging.source_interface }}
 {%     for match_list in logging.policy.match.match_lists | arista.avd.natural_sort('name') %}
 logging policy match match-list {{ match_list.name }} {{ match_list.action }}
 {%     endfor %}
+{%     if logging.level is arista.avd.defined %}
+{%         for level in logging.level | arista.avd.natural_sort('facility_name') %}
+{%             set logging_cli = "logging" %}
+{%             set logging_cli = logging_cli ~ " level " ~ level.facility_name ~ " " ~ level.severity %}
+{{ logging_cli }}
+{%         endfor %}
+{%     endif %}
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2]:
```suggestion
{%     for level in logging.level | arista.avd.natural_sort('facility_name') %}
{%         if level.severity is arista.avd.defined %}
logging level {{ level.facility_name }} {{ level.severity }}
{%         endif %}
{%     endfor %}
```

**Review by Shivani-gslab — CHANGES_REQUESTED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml [hunk 2]
```diff
@@ -149,3 +149,47 @@ keys:
...
+        description: Configure logging severity.
+        primary_key: facility_name
+        convert_types:
+          - dict
+        items:
+          type: dict
+          keys:
+            facility_name:
+              type: str
+              description: Facility name.
```
Comment (Shivani-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml]:
```suggestion
```

Comment (Shivani-gslab) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml]: No need to add the description here as the key name itself tells the same.

**Review by Shivani-gslab — COMMENTED** [review]
Please update the documentation template also.

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/logging.cfg [hunk 3]
```diff
@@ -19,6 +19,9 @@ logging format timestamp traditional year timezone
 logging format rfc5424
 logging source-interface Loopback0
 logging vrf mgt source-interface Management0
+logging level AAA warnings
+logging level ACL critical
+logging level BGP 0
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/logging.cfg:31]: In CLI the output changes to `logging level BGP emergencies`.
May be we can map the numbers with its corresponding values in J2 file (using dict).

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2 [hunk 4]
```diff
@@ -93,4 +93,9 @@ logging source-interface {{ logging.source_interface }}
 {%     for match_list in logging.policy.match.match_lists | arista.avd.natural_sort('name') %}
 logging policy match match-list {{ match_list.name }} {{ match_list.action }}
 {%     endfor %}
+{%     for level in logging.level | arista.avd.natural_sort('facility_name') %}
+{%         if level.severity is arista.avd.defined %}
```
Comment (Vibhu-gslab) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2:97]: Something like..
```
{%             set severity_mapping = {'0': 'emergencies', '1': 'alerts'} %}
{%             set severity = level.severity %}
{%             if severity in severity_mapping %}
{%                 set severity = severity_mapping[severity] %}
{%             endif %}
logging level {{ level.facility_name }} {{ severity }}
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/logging.cfg (hunk 3 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/logging.cfg:31]: I would prefer not to have such mapping in `eos_cli_config_gen`. Just add valid values for both the names and the numbers and then let the user decide.

Comment (Vibhu-gslab) [reply ansible_collections/arista/avd/molecule/eos_cli_config_gen/intended/configs/logging.cfg:31]: Okay :)

**Review by Vibhu-gslab — APPROVED** [review]

**Review by Shivani-gslab — APPROVED** [review]
LGTM

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml [hunk 5]
```diff
@@ -149,3 +149,46 @@ keys:
...
+              description: |-
+                Severity of facility. Below are the supported severites.
+                alerts         Immediate action needed           (severity=1)
+                critical       Critical conditions               (severity=2)
+                debugging      Debugging messages                (severity=7)
+                emergencies    System is unusable                (severity=0)
+                errors         Error conditions                  (severity=3)
+                informational  Informational messages            (severity=6)
+                notifications  Normal but significant conditions (severity=5)
+                warnings       Warning conditions                (severity=4)
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml]:
```suggestion
                emergencies    System is unusable                (severity=0)
                alerts         Immediate action needed           (severity=1)
                critical       Critical conditions               (severity=2)
                errors         Error conditions                  (severity=3)
                warnings       Warning conditions                (severity=4)
                notifications  Normal but significant conditions (severity=5)
                informational  Informational messages            (severity=6)
                debugging      Debugging messages                (severity=7)
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/logging.md [hunk 6]
```diff
@@ -70,6 +70,12 @@ interface Management1
 | mgt | 40.40.40.7 | 300, 400 | UDP |
 | vrf_with_no_source_interface | 1.2.3.4 | Default | UDP |
 
+| Level Facility | Severity |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/logging.md]:
```suggestion
| Facility | Severity |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2 [hunk 7]
```diff
@@ -69,6 +69,16 @@
 {%             endif %}
 {%         endfor %}
 {%     endif %}
+{%     if logging.level is arista.avd.defined %}
+
+| Level Facility | Severity |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2]:
```suggestion
| Facility | Severity |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2 [hunk 8]
```diff
@@ -69,6 +69,16 @@
...
+| -------------- | -------- |
+{%         for level in logging.level | arista.avd.natural_sort('facility_name') %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2]:
```suggestion
{%         for level in logging.level | arista.avd.natural_sort('facility') %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2 [hunk 9]
```diff
@@ -69,6 +69,16 @@
...
+{%             if level.severity is arista.avd.defined %}
+| {{ level.facility_name }} | {{ level.severity }} |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2]:
```suggestion
| {{ level.facility }} | {{ level.severity }} |
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2 [hunk 10]
```diff
@@ -93,4 +93,9 @@ logging source-interface {{ logging.source_interface }}
...
+logging level {{ level.facility_name }} {{ level.severity }}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2]:
```suggestion
logging level {{ level.facility }} {{ level.severity }}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2 [hunk 11]
```diff
@@ -93,4 +93,9 @@ logging source-interface {{ logging.source_interface }}
 {%     for match_list in logging.policy.match.match_lists | arista.avd.natural_sort('name') %}
 logging policy match match-list {{ match_list.name }} {{ match_list.action }}
 {%     endfor %}
+{%     for level in logging.level | arista.avd.natural_sort('facility_name') %}
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/logging.j2]:
```suggestion
{%     for level in logging.level | arista.avd.natural_sort('facility') %}
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml [hunk 12]
```diff
@@ -149,3 +149,46 @@ keys:
...
+      level:
+        type: list
+        description: Configure logging severity.
+        primary_key: facility_name
+        convert_types:
+          - dict
+        items:
+          type: dict
+          keys:
+            facility_name:
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml]:
```suggestion
            facility:
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml [hunk 13]
```diff
@@ -149,3 +149,46 @@ keys:
                     min: 10
                     max: 65535
                     description: Logging interval in seconds.
+      level:
+        type: list
+        description: Configure logging severity.
+        primary_key: facility_name
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/schemas/schema_fragments/logging.schema.yml]:
```suggestion
        primary_key: facility
```

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2 [hunk 14]
```diff
@@ -69,6 +69,16 @@
 {%             endif %}
 {%         endfor %}
 {%     endif %}
+{%     if logging.level is arista.avd.defined %}
+
+| Facility | Severity |
+| -------------- | -------- |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/logging.j2]:
```suggestion
| Facility | Severity |
| -------- | -------- |
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/logging.md [hunk 15]
```diff
@@ -70,6 +70,12 @@ interface Management1
 | mgt | 40.40.40.7 | 300, 400 | UDP |
 | vrf_with_no_source_interface | 1.2.3.4 | Default | UDP |
 
+| Facility | Severity |
+| -------------- | -------- |
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/documentation/devices/logging.md]:
```suggestion
| Facility | Severity |
| -------- | -------- |
```

File: ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/logging.yml [hunk 16]
```diff
@@ -71,3 +71,10 @@ logging:
       discards:
         global: true
         interval: 10
+  level:
+    - facility_name: BGP
+      severity: 0
+    - facility_name: AAA
+      severity: warnings
+    - facility_name: ACL
+      severity: critical
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/molecule/eos_cli_config_gen/inventory/host_vars/logging.yml]:
```suggestion
    - facility: BGP
      severity: 0
    - facility: AAA
      severity: warnings
    - facility: ACL
      severity: critical
```

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**